docs/sources/operators-guide/configure/reference-configuration-parameters/index.md

```diff
@@ -2448,6 +2448,10 @@ The `limits` block configures default and per-tenant limits imposed by component
 # rules.
 [forwarding_endpoint: <string> | default = ""]
 
+# If set, forwarding drops samples that are older than this duration. If unset
+# or 0, no samples get dropped.
+[forwarding_drop_before: <int> | default = ]
```

Comment:
Can we make it display `<duration>`? I tried it with this patch, and it worked, although I'm not sure which of the cases really helped:

```patch
diff --git a/tools/doc-generator/parse/parser.go b/tools/doc-generator/parse/parser.go
index 293de6790..bff59d0a0 100644
--- a/tools/doc-generator/parse/parser.go
+++ b/tools/doc-generator/parse/parser.go
@@ -342,6 +342,8 @@ func getFieldCustomType(t reflect.Type) (string, bool) {
                return "url", true
        case reflect.TypeOf(time.Duration(0)).String():
                return "duration", true
+       case reflect.TypeOf(model.Duration(0)).String():
+               return "duration", true
        case reflect.TypeOf(flagext.StringSliceCSV{}).String():
                return "string", true
        case reflect.TypeOf(flagext.CIDRSliceCSV{}).String():
@@ -422,6 +424,8 @@ func getCustomFieldType(t reflect.Type) (string, bool) {
                return "url", true
        case reflect.TypeOf(time.Duration(0)).String():
                return "duration", true
+       case reflect.TypeOf(model.Duration(0)).String():
+               return "duration", true
        case reflect.TypeOf(flagext.StringSliceCSV{}).String():
                return "string", true
        case reflect.TypeOf(flagext.CIDRSliceCSV{}).String():
```

Reply:
nice, thanks, will apply that

Reply:
It leads to panic when generating config descriptor json file:

```
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x101150a40]

goroutine 1 [running]:
github.com/grafana/mimir/pkg/mimirtool/config.(*InspectedEntry).decodeValue(0x140010fb600, {0x101870760, 0x140010fcb40})
	/Users/peter/Grafana/mimir/pkg/mimirtool/config/inspect.go:284 +0x120
github.com/grafana/mimir/pkg/mimirtool/config.parseDefaultValue(0x140010fb600, {0x0, 0x0})
	/Users/peter/Grafana/mimir/pkg/mimirtool/config/inspect.go:539 +0x108
github.com/grafana/mimir/pkg/mimirtool/config.convertEntryToEntry(0x14000ee36b0)
	/Users/peter/Grafana/mimir/pkg/mimirtool/config/inspect.go:525 +0x1f8
github.com/grafana/mimir/pkg/mimirtool/config.convertEntriesToEntries(...)
	/Users/peter/Grafana/mimir/pkg/mimirtool/config/inspect.go:497
github.com/grafana/mimir/pkg/mimirtool/config.convertEntryToEntry(0x14000eba9a0)
	/Users/peter/Grafana/mimir/pkg/mimirtool/config/inspect.go:517 +0x50c
github.com/grafana/mimir/pkg/mimirtool/config.convertEntriesToEntries(...)
	/Users/peter/Grafana/mimir/pkg/mimirtool/config/inspect.go:497
github.com/grafana/mimir/pkg/mimirtool/config.convertBlockToEntry(...)
	/Users/peter/Grafana/mimir/pkg/mimirtool/config/inspect.go:547
github.com/grafana/mimir/pkg/mimirtool/config.InspectConfigWithFlags({0x101792aa0?, 0x14000d20000?}, 0x1018713e0?)
	/Users/peter/Grafana/mimir/pkg/mimirtool/config/inspect.go:491 +0xe0
github.com/grafana/mimir/pkg/mimirtool/config.InspectConfig({0x10186d280?, 0x14000d20000})
	/Users/peter/Grafana/mimir/pkg/mimirtool/config/inspect.go:478 +0x5c
github.com/grafana/mimir/pkg/mimirtool/config.Describe({0x10186d280?, 0x14000d20000?})
	/Users/peter/Grafana/mimir/pkg/mimirtool/config/inspect.go:468 +0x24
main.main()
	/Users/peter/Grafana/mimir/tools/config-inspector/main.go:14 +0x34
exit status 2
```

Let's tackle that separately.